magda-search-api/src/main/scala/au/csiro/data61/magda/api/SearchApi.scala

```diff
@@ -38,6 +83,57 @@ class SearchApi(val searchQueryer: SearchQueryer)(implicit val config: Config, i
               }
           }
         } ~
+        /**
+         * @apiGroup Search
+         * @api {get} /v0/search/datasets Search Datasets
+         * @apiDescription Returns a list of results.
+         *
+         * @apiParam (Query) {string} [query] full text search query
+         * @apiParam (Query) {number} [start=0] index of first result to return
+         * @apiParam (Query) {number} [limit=10] number of results to return
+         * @apiParam (Query) {number} [facetSize=10] number of facets to return
+         * @apiParam (Query) {string[]} [publisher] filter search query by names of organisations
+         * @apiParam (Query) {string} [dateFrom] filter datasets by start date of dataset coverage
+         * @apiParam (Query) {string} [dateTo] filter datasets by end date of dataset coverage
+         * @apiParam (Query) {string[]} [region] filter datasets by regions
+         * @apiParam (Query) {string[]} [format] filter datasets by formats
+         *
+         * @apiSuccess {object} query Will reflect query specified.
+         * @apiSuccess {string} hitCount number of total results.
+         * @apiSuccess {Dataset[]} dataSets Result datasets.
+         * @apiSuccess {object} temporal Reflects da
+         * @apiSuccess {object} temporal.start
+         * @apiSuccess {string} temporal.start.date Returns the start date of the earliest matched result in ISO8601 format.
+         * @apiSuccess {object} temporal.end
+         * @apiSuccess {string} temporal.end.date Returns the end date of the latest matched result in ISO8601 format.
+         * @apiSuccess {string} strategy search strategy used
+         * @apiSuccess {Facets} facets Facets of results. See response of Get Facet Options for more details.
+         *
+         * @apiSuccessExample {any} 200
+         *    {
+         *        "hitCount": 948,
+         *        "dataSets": [
+         *            {
+         *                "quality": 0.6,
+         *                "identifier": "...",
+         *                ...
```

Comment:
There needs to be some doco around what a `Dataset` looks like, either an example or more details in `@apiSuccess`.